src/main/kotlin/org/jetbrains/fortran/lang/preprocessor/FortranPreprocessingLexer.kt

```diff
@@ -19,20 +20,45 @@ class FortranPreprocessingLexer : LookAheadLexer(FortranLexer(false)) {
         macrosContext.isDefined(identifier)
     }
     private val ifNotDefinedDecisionEvaluator = { name: CharSequence -> !ifDefinedDecisionEvaluator(name) }
+    private val ifDecisionEvaluator = { directive_content: CharSequence -> !directive_content.endsWith("/* macro_eval: false */") }
 
     override fun lookAhead(baseLexer: Lexer) {
+        val CONDITION_DIRECTIVES = TokenSet.create(IF_DIRECTIVE, IF_DEFINED_DIRECTIVE, IF_NOT_DEFINED_DIRECTIVE, ELSE_DIRECTIVE, ELIF_DIRECTIVE, ENDIF_DIRECTIVE)
```

Comment:
This `CONDITION_DIRECTIVES` should be refactored out. 

I didn't get why, but several token sets are empty: `FortranTokenType.DIRECTIVES`,  `FortranTokenType.IF_DIRECTIVES`, `FortranTokenType.END_IF_DIRECTIVES`